.github/ISSUE_TEMPLATE/bug_report.yml

```diff
@@ -0,0 +1,38 @@
+name: I think I found a bug in SQLx
+description: Create a bug-report issue
+labels: [bug]
+body:
+  - type: textarea
+    id: related-issues
+    validations:
+      required: true
+    attributes:
+      label: I have found these related issues/pull requests
+      description: "I have searched by clicking [HERE](https://github.com/launchbadge/sqlx/issues?q=) for existing issues, these are the ones I've found, and this is why I think this deserves a new issue."
+      placeholder: "Related to #1 by also touching the ... system. They should not be merged because ..."
+  - type: textarea
+    id: description
+    validations:
+      required: true
+    attributes:
+      label: Description
+      description: Clear and concise description of what the bug is
+  - type: textarea
+    id: steps-to-reproduce
+    validations:
+      required: true
+    attributes:
+      label: Reproduction steps
+      description: A small code snippet or a link to a Github repo or Gist, with instructions on reproducing the bug.
+  - type: textarea
+    id: deployment-info
+    attributes:
+      label: Environment Information
+      value: |
+        * SQLx version: [REQUIRED]
+        * SQLx features enabled: [REQUIRED]
+        * Database server and version: [REQUIRED] (MySQL / Postgres / SQLite <x.y.z>)
+        * Operating system: [REQUIRED]
+        * `rustc --version`: [REQUIRED]
```

Comment:
Might as well break these out into separate fields.

Reply:
Fair enough.
This has the downside of looking a bit longer in the issue that is genereated, but that is likely an ok downside